### ISSUE TITLE
Possible bug with dynamic_intervals?

### ISSUE THREAD

**Issue by AndreaPi** [open-mmlab/mmdetection#10543]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
I'm training a RTMDet model. I didn't specify `dynamic_intervals` in my `train_cfg`, but when I print the whole config with `print_config.py`, it's there:

```
train_cfg = dict(
    type='EpochBasedTrainLoop',
    max_epochs=300,
    val_interval=1,
    dynamic_intervals=[(280, 1)])
```
If I understand the [documentation](https://mmdetection.readthedocs.io/en/latest/advanced_guides/customize_runtime.html#customize-train-loop) correctly, this means that the validation loss should be printed every 280 epochs for the first 280 epochs, and then every epoch. Is that correct? However, it seems to me that it's printed at each epoch (which makes more sense):

![image](https://github.com/open-mmlab/mmdetection/assets/8233615/46300fc9-4fe1-4439-9976-ef1c4e199afd)

**Reproduction**

1. What command or script did you run?

```none
python /mmdetection/tools/train.py \
        "./configs/simple_rtmdet_s_8xb32-300e.py"
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
Yes, here is the config file:
```
_base_ = [
    '/mmdetection/configs/rtmdet/rtmdet_s_8xb32-300e_coco.py',
    ]

experiment_id = 'simple_rtmdet_s_8xb32-300e'


classes = ('A', 'B')
num_classes = len(classes)
metainfo = {
    'classes':
        classes,
    'palette':
        [(190, 77, 37), (37, 150, 190)]
}

model = dict(
    data_preprocessor=dict(
        type='DetDataPreprocessor',
        mean=[213.308, 102.466, 48.287],
        std=[31.949, 31.084, 20.839],
        bgr_to_rgb=False,
        batch_augments=None,
        pad_size_divisor=32),
    bbox_head=dict(num_classes=num_classes)
)

image_scale = (512, 672)
batch_size = 4
n_workers = 4
n_gpu = 1
base_lr = 0.004
eta = base_lr * (batch_size * n_gpu / 16)
n_epochs = 300

# pipelines setting
packed_inputs_items = ('img_id', 'img_path', 'img_shape', 'scale_factor', 'ori_shape')
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', scale=image_scale, keep_ratio=True),
    dict(type='PackDetInputs', meta_keys=packed_inputs_items),
]
val_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', scale=image_scale, keep_ratio=True),
    dict(type='PackDetInputs', meta_keys=packed_inputs_items),
]
test_pipeline = val_pipeline

# dataset settings
dataset_type = 'CocoDataset'
data_root = '/data/xxx/'
train_annotations = 'annotations/train.json'
val_annotations = 'annotations/val.json'
test_annotations = 'annotations/test.json'
image_data_prefix = data_root + 'images/'
backend_args = None

workers_configs = {
    "num_workers": n_workers,
    "persistent_workers": True,
}
dataset_configs = {
    "type": dataset_type,
    "metainfo": metainfo,
    "data_root": data_root,
    "data_prefix": dict(img=image_data_prefix),
    "backend_args": backend_args
}

# dataloader settings
train_dataloader = dict(
    batch_size=batch_size,
    sampler=dict(type='DefaultSampler', shuffle=True),
    **workers_configs,
    dataset=dict(
        ann_file=train_annotations,
        filter_cfg=dict(filter_empty_gt=True, min_size=32),
        pipeline=train_pipeline,
        **dataset_configs
    )
)
val_dataloader = dict(
    batch_size=1,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    **workers_configs,
    dataset=dict(
        ann_file=val_annotations,
        pipeline=val_pipeline,
        test_mode=True,
        **dataset_configs)
)
test_dataloader = dict(
    batch_size=1,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    **workers_configs,
    dataset=dict(
        ann_file=test_annotations,
        pipeline=test_pipeline,
        test_mode=True,
        **dataset_configs)
)

# evaluator settings
val_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + val_annotations,
    metric='bbox',
    format_only=False,
    backend_args=backend_args
)
test_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + test_annotations,
    metric='bbox',
    format_only=False,
    backend_args=backend_args
)

# optimizer settings
optimizer = dict(
    type='AdamW',
    lr=eta,
    weight_decay=0.0001
)
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=optimizer,
)

# workflow and workdir settings
train_cfg = dict(
    type='EpochBasedTrainLoop',
    max_epochs=n_epochs,
    val_interval=1)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')

# checkpoint: save the best model
default_hooks = dict(
    checkpoint=dict(
        type='CheckpointHook',
        interval=-1,
        save_best='auto'),
    logger=dict(
        type='LoggerHook',
        interval=50)
)

workflow = [('train', 1), ('val', 1)]

work_dir = f'logs/{experiment_id}'

# logger settings
vis_backends = [
    dict(type='LocalVisBackend'),
    dict(type='TensorboardVisBackend'),
]
visualizer = dict(
    type='DetLocalVisualizer',
    vis_backends=vis_backends,
    name='visualizer'
)
```

4. What dataset did you use?
A proprietary one, in COCO format, with only 2 classes.

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
```
sys.platform: linux
Python: 3.7.10 (default, Feb 26 2021, 18:47:35) [GCC 7.3.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: NVIDIA A100-SXM4-40GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.1, V11.1.105
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.9.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.10.0
OpenCV: 4.7.0
MMEngine: 0.7.3
MMDetection: 3.0.0+

```

3. You may add addition that may be helpful for locating the problem, such as
   N/A
**Error traceback**
N/A
